### ISSUE TITLE
cmake parsing KERNEL* files incorrectly

### ISSUE THREAD

**Issue by mistomin** [xianyi/OpenBLAS#2583]
Currently cmake function `ParseMakefileVars` (which is used for parsing `KERNEL*` files) skips lines with `if`, `ifeq`, `ifneq` and just converts all assignments in config file to cmake variable assignments. It leads to execution of both `if` and `else` branches. For example in
```
ifeq($(DGEMM_UNROLL_M), 8)
DGEMMINCOPY = a
else
DGEMMINCOPY = b
endif
```
cmake just sets DGEMMINCOPY to `a` and then immediately resets in to `b`. This means wrong kernel will be called. And this is common pattern in `KERNEL*` files to use such `if-else`. So I have a lot of examples, when variable is reset to wrong value and it brokes performance or correctness. Often it is reference version instead of optimized. But sometimes it leads to segfaults.

**Comment by martin-frbg** [comment]
Known defect unfortunately. The KERNEL files are basically Makefiles in disguise, so the gmake handling of conditionals would need to be reimplemented in ParseMakefileVars (which currently only knows to extract key-value pairs).

**Comment by martin-frbg** [comment]
First attempt at a fix in #2584 - but ~~requires an else branch to be present and~~ does not yet support nested conditionals

**Comment by martin-frbg** [comment]
Actually this appears to work quite well even without explicit support for nested conditionals.
